### DISCUSSION TITLE
Creating an EXE from a Streamlit app using cx_Freeze on obfuscated scripts by pyarmor

### DISCUSSION THREAD

**Discussion by Waschbrettwade** [marcelotduarte/cx_Freeze#2406] [Q&A]
Hi there,

I have successfully turned my dummy Streamlit app into an EXE using cx_Freeze. However, I also want to protect my code. That's why I also obfuscate my scripts using pyarmor. Now, whether I first obfuscate my scripts and then create the exe, or the other way around, something doesn't quite work. I get the EXE to correctly execute at first. But when I press any button in the Streamlit app, I get 

```
	RuntimeError: the format of obfuscated script is incorrect (1:1082)
	Traceback:

	File "C:\Users\user\Python\Installer cx_freeze test 2\.venv\Lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 600, in _run_script
	exec(code, module.__dict__)
	File "<frozen __main__>", line 3, in <module>
```

When using obfuscated scripts without cx_Freeze, there is no problem. Does anyone have an idea what's causing this and how to fix this? Thank you very much!

**Comment by luigirovani** [comment]
When you try to freeze an obfuscated Python script, freeze is unable to find your script's dependencies, only pyarmor_runtime. The way I solved this was by adding an extra script for freezing, which only imports the dependencies your code needs (you can try doing this by copying and including all the imports you make in the plain script).

Declare it along with setup.py to be freezing together. After build, you could delete it; it will serve so that ex_freeze can track its dependencies.

**Reply by Waschbrettwade** [reply]
Thank you for your answer. However, this seems to be unrelated to my problem as the initial start works, but it collapses and throws the stated error after a Streamlit rerun (which occurs when something is clicked) that runs the scripts anew top down from the entry point on.

**Comment by abaxi** [comment]
@Waschbrettwade could you please share the steps you took to make an executable for streamlit? 

I can't get a simple "hello world" streamlit application converted into an EXE. The moment I run any of the following commands, the command prompt gets stuck after printing a few lines of progress


main.py has just 2 lines:
```
import streamlit as st
st.header("Hello world")
```


Using python 3.13, streamlit 1.51, cxfreeze 8.4.1,
```
PS C:\Users\ab\pyprojects\example> cxfreeze-quickstart.exe
Project name: main
Version [1.0]:
Description: test streamlit
Python file to make executable from: main.py
Executable file name [main]:
(C) console application, (G) GUI application, or (S) service [C]: G
Save setup script to [setup.py]:
Overwrite setup.py [n]? y

Setup script written to setup.py; run it as:
    python setup.py build
Run this now [n]? y
running build
running build_exe

```

After a few minutes of no change, when I cancel the run, I get the following error which makes it seem like the system is just waiting for something.

```
interrupted
Traceback (most recent call last):
  File "C:\Users\ab\miniconda3\Scripts\cxfreeze-quickstart-script.py", line 10, in <module>
    sys.exit(main())
             ~~~~^^
  File "C:\Users\ab\miniconda3\Lib\site-packages\cx_Freeze\setupwriter.py", line 126, in main
    subprocess.call([sys.executable, writer.setup_file_name, "build"])
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\ab\miniconda3\Lib\subprocess.py", line 397, in call
    return p.wait(timeout=timeout)
           ~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\ab\miniconda3\Lib\subprocess.py", line 1280, in wait
    return self._wait(timeout=timeout)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\ab\miniconda3\Lib\subprocess.py", line 1606, in _wait
    result = _winapi.WaitForSingleObject(self._handle,
                                         timeout_millis)
KeyboardInterrupt
```
